docs/basics/101-146-providers.rst

```diff
@@ -0,0 +1,86 @@
+.. _providers:
+
+Configuring new data providers
+------------------------------
+
+DataLad can download files via the ``http``, ``https``, ``ftp``, and ``s3``
+protocol. If data retrieval requires authentication, DataLad's provides an interface to
+query, request, and store the most common type of credentials. An example of this
+is shown in the usecase :ref:`usecase_HCP_dataset`: :command:`datalad get` will
+ask for AWS S3 credential to retrieve data that is stored in S3 buckets from
+the command line once, and -- given valid credentials -- download the files
+and all future files from this bucket right away. There
+are a number of natively supported types of authentication (a list is given in
+the hidden section below) and out-of-the box access to a broad range
+of providers such as `S3 <https://aws.amazon.com/s3/?nc1=h_ls>`_ or
+`LORIS <https://loris.ca/>`_.
+If you attempt to retrieve data from supported providers that require authentication,
+you will be prompted for your credentials from the command line at the time of
+the first download, and subsequent downloads handle authentication in the background.
+
+If a provider requires authentication but it is not known to DataLad yet, the
+download will fail. However, the provider can be configured by
+the user to enable interactions with it.
+With a provider configuration in place, commands such as :command:`datalad download-url`
+or :command:`datalad add-urls` can work with urls of custom providers, and
+:command:`datalad get` is enabled to retrieve file contents from these sources.
+The configuration can either be done in the terminal upon a prompt from the
+command line when a download fails due to a missing provider configuration,
+or by placing provider-specific configuration files into ``.datalad/providers``.
+
+In order to configure a provider configuration, one needs to input a
+provider name, a regular expression that can match a url one would want to download
+from, an authentication type, and a credential type. The example below sheds some
+light one this.
+
+.. findoutmore:: Which authentication and credential types are possible?
+
+   The following credential types are supported:
+
+   - ``'user_password'``
+   - ``'aws-s3'``
+   - ``'nda-s3'``
+   - ``'token'``
+   - ``'loris-token'``
```

Comment:
I would remove nda-s3 and loris, and limit to the rest, while referring to them as common credential types, possibly mention that others exist. This will communicate that this list does not claim to be comprehensive. The documentation of that functionality in datalad should provides such a list, which could be referenced here, if one is available.